framework-cloud/framework-openshift-apb/src/main/java/org/kie/cloud/openshift/scenario/ClusteredWorkbenchRuntimeSmartRouterTwoKieServersTwoDatabasesScenarioApb.java

```diff
@@ -0,0 +1,237 @@
+        return kieServerOneDeployment;
+    }
+
+    @Override
+    public KieServerDeployment getKieServerTwoDeployment() {
+        return kieServerTwoDeployment;
+    }
+
+    @Override
+    public DatabaseDeployment getDatabaseOneDeployment() {
+        return databaseOneDeployment;
+    }
+
+    @Override
+    public DatabaseDeployment getDatabaseTwoDeployment() {
+        return databaseTwoDeployment;
+    }
+
+    @Override
+    public List<Deployment> getDeployments() {
+        List<Deployment> deployments = new ArrayList<Deployment>(Arrays.asList(workbenchRuntimeDeployment, smartRouterDeployment, kieServerOneDeployment, kieServerTwoDeployment, databaseOneDeployment, databaseTwoDeployment));
+        deployments.removeAll(Collections.singleton(null));
+        return deployments;
+    }
+
+    private WorkbenchDeployment createWorkbenchRuntimeDeployment(Project project) {
+        WorkbenchRuntimeDeploymentImpl workbenchRuntimeDeployment = new WorkbenchRuntimeDeploymentImpl(project);
+        workbenchRuntimeDeployment.setUsername(DeploymentConstants.getWorkbenchUser());
+        workbenchRuntimeDeployment.setPassword(DeploymentConstants.getWorkbenchPassword());
+
+        return workbenchRuntimeDeployment;
+    }
+
+    private SmartRouterDeployment createSmartRouterDeployment(Project project) {
+        SmartRouterDeploymentImpl smartRouterDeployment = new SmartRouterDeploymentImpl(project);
+
+        return smartRouterDeployment;
+    }
+
+    private KieServerDeploymentImpl createKieServerDeployment(Project project, String kieServerIndex) {
+        KieServerDeploymentImpl kieServerDeployment = new KieServerDeploymentImpl(project);
+        kieServerDeployment.setUsername(DeploymentConstants.getKieServerUser());
+        kieServerDeployment.setPassword(DeploymentConstants.getKieServerPassword());
+        kieServerDeployment.setServiceSuffix("-" + kieServerIndex);
+
+        return kieServerDeployment;
+    }
+
+    private DatabaseDeploymentImpl createDatabaseDeployment(Project project, String databaseIndex) {
+        DatabaseDeploymentImpl databaseDeployment = new DatabaseDeploymentImpl(project);
+        databaseDeployment.setServiceSuffix("-" + databaseIndex);
+        return databaseDeployment;
+    }
+
+    @Override
+    public List<WorkbenchDeployment> getWorkbenchDeployments() {
+        return Arrays.asList(workbenchRuntimeDeployment);
+    }
+
+    @Override
+    public List<KieServerDeployment> getKieServerDeployments() {
+        return Arrays.asList(kieServerOneDeployment, kieServerTwoDeployment);
+    }
+
+    @Override
+    public List<SmartRouterDeployment> getSmartRouterDeployments() {
+        return Arrays.asList(smartRouterDeployment);
+    }
+
+    @Override
+    public List<ControllerDeployment> getControllerDeployments() {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public SsoDeployment getSsoDeployment() {
+        throw new UnsupportedOperationException("Not supported yet.");
```

Comment:
Is there a reason why SSO deployment is not returned? It seems that it can be configured for this scenario.

Reply:
It's not possible to configure SSO for this scenario. We are not able to create SSO client for more then 1 set of Kie Servers.